### PR TITLE
Add displaying line number on failing test with tap output

### PR DESCRIPTION
**Pull request by winksaville**
# Description

# Checklist

- [x] I am an awesome developer and proud of my code
- [x] I added / updated / removed relevant unit or integration tests to prove my change works
- [x] I ran all tests using ```npm test``` to make sure everything else still works
- [x] I ran ```npm run review``` to ensure the code adheres to the repository standards

There is one error when running ```npm run review``` in `core/tracing.ts` which I don't know how to fix, please advise:
```
core/tracing.ts[1, 1]: require statement not part of an import statement
```

# Additional Information

### CODE REVIEW

File: core/tracing.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
+require("source-map-support").install(); // tslint:disable-line
```
Comment (winksaville) [inline core/tracing.ts]: Here is the hack to get the tslint error, any suggestions on how to use "import"?

**Comment by winksaville** [comment]
[Added more tests](https://github.com/alsatian-test/alsatian/pull/371/commits/506e6a99d12adbe7f776b31f61b73fc3a9f527d2) but still not quite 100% coverage. Any clues on why TypeMatchError isn't 100% on branches?

File: core/tracing.ts (hunk 1 above)
Comment (Jameskmonger) [inline core/tracing.ts]: @winksaville 

```
import * as SourceMapSupport from "source-map-support";

SourceMapSupport.install();
```

**Comment by winksaville** [comment]
Wow, finding the 50% coverage problem was a hard, I figured it out by looking at the unit-test with --tap and seeing RegexMatchError in the `temporary-branch-fix.spec.ts` output.

**Comment by winksaville** [comment]
So this is kinda working, except it's failing the gulp integration-tests on Appveyor but on Travis the gulp is working but the babel integration-tests fail. The problem is that the line and column number of the "expected-output" files are different from my machine so the `toContain` tests fail. I also see different line/column numbers on my machine depending on what I do and I'm not sure exactly what causes it.

A couple of possible solutions off the top of my head:
- Allow passing run time flag to Alsatian which inhibits line and column numbers
- If the problem is related to different run time versions of code, then using yarn instead of npm to install dependencies might fix the problem.

I use yarn here, although all of the stuff I do is simple compared to Alsatian I've found it pretty good. Tomorrow I'll give it a try and if it operates I'd suggest switching even if it doesn't solve the problem I'm seeing.

**Comment by winksaville** [comment]
@jamesrichford, the failure in Travis is a runtime syntax error:
```
> node ./cli/alsatian-cli.js "./test/unit-tests/**/*.spec.js"
/home/travis/build/alsatian-test/alsatian/node_modules/traceloc/out/traceloc.js:35
function here(callDepth = 0) {
                        ^
SyntaxError: Unexpected token =
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:373:25)
    at Module.replacementCompile (/home/travis/build/alsatian-test/alsatian/node_modules/nyc/node_modules/append-transform/index.js:58:13)
    at module.exports (/home/travis/build/alsatian-test/alsatian/node_modules/nyc/node_modules/default-require-extensions/js.js:8:9)
    at Object.<anonymous> (/home/travis/build/alsatian-test/alsatian/node_modules/nyc/node_modules/append-transform/index.js:62:4)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/home/travis/build/alsatian-test/alsatian/core/expect.js:1:51220)
```
I'm guessing the code being generated isn't compatible with the V8 engine, but just a guess.

I'd like to suggest we move away from supporting Node 4, in the [Node 6 blog announcement](https://nodejs.org/en/blog/release/v6.0.0/) from April last year it says:

"In accordance with our Long Term Support plan, Node.js v4 'Argon' will continue to be supported as an Active Long Term Support stream until April of 2017, after which it will transition into what we call Maintenance LTS."

I'll create a PR to remove it and additional discussion on that subject can happen there.

**Comment by winksaville** [comment]
About time, everything passed!!

**Comment by winksaville** [comment]
Rebased on to lastest master.

**Pull request closed without merging**